src/main/java/org/mule/service/http/impl/service/client/CursorInputStreamBodyGenerator.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.service.http.impl.service.client;
+
+import java.io.IOException;
+
+import org.mule.runtime.api.streaming.bytes.CursorStream;
+
+import com.ning.http.client.Body;
+import com.ning.http.client.generators.InputStreamBodyGenerator;
+
+/**
+ * Input Stream Body Generator which properly resets input streams
+ * 
+ * @since 4.3.0
+ *
+ */
+public class CursorInputStreamBodyGenerator extends InputStreamBodyGenerator {
+
+  public CursorInputStreamBodyGenerator(CursorStream inputStream) {
+    super(inputStream);
+  }
+
+  @Override
+  public Body createBody() throws IOException {
+    if (!inputStream.markSupported()) {
+      ((CursorStream) inputStream).seek(0);
```

Comment:
Why doesn't it catch the seek(0) exception as in CursorNonBlockingInputStreamFeeder#reset() ?